docs/book/14-Streams.md

```diff
@@ -7,7 +7,7 @@
 
 流是一系列与特定存储机制无关的元素——实际上，流并没有“存储”之说。
 
-利用流，我们无需迭代集合中的元素，就可以提取和操作它们。这些管道通常被组合在一起，在流上形成一条操作管道。
+使用流可以从管道提取和操作元素，而不用在集合中迭代元素。这些管道通常被组合在一起，形成一系列对流进行操作的管道。
```

Comment:
这句话你真的觉得改了语序变好了？

Reply:
原文：Instead of iterating through elements in a collection, with a stream you draw elements from a pipe and operate on them. These pipes are typically strung together to form a pipeline of operations upon the stream.
改后的翻译文字更多，但是更清晰易懂。这句也可以保持原翻译，不改也行。

Reply:
使用流可以从管道提取和操作元素，而不用在集合中迭代元素

这种语序在中文中其实不太好。

「利用流，我们无需迭代集合中的元素，就可以提取和操作它们」
「使用流可以从管道提取和操作元素，而不用在集合中迭代元素」
这两句话，想想自己平常阅读的中文书籍的语序。看看哪个更好

Reply:
这句话两种翻译都可以，先说流自己的特点，然后说流避免了哪些缺点，我感觉这样更符合介绍事物的一般顺序。原翻译是按照英文语序翻的，英语原文其实是把一个短句前置了。我这个提交你有修改权，觉得需要改的地方直接改就行。我自己的版本是按照我修改的，发的pull request只是个建议，因为原翻译确实有好多地方不好读。

Reply:
给我的感觉，这句话没有说避免了哪些缺点。

都是在介绍「声明式」编程的特点。而且这个特点是具有递进性的。

Reply:
那就保留原翻译

Reply:
也改进一下吧
使用流，无需在迭代集合中的元素，就可以从管道提取和操作元素

Reply:
可以。其实原翻译有歧义，“利用流，我们无需迭代集合中的元素，就可以提取和操作它们。”意思容易被误解为：利用流，我们无需迭代集合中的元素，就能从集合中提取和操作它们。
而且原文作者一直说的是“you”，原翻译一直在说“我们”。

Reply:
这种翻译主题倒置的方法看「第四版」翻译也挺多的

揣测的目的是读起来更有代入性吧。

书里面的主语一直都在说「你」「你」可以 xxx 样，确实有不如「我们」这种代词更亲切

Reply:
java编程思想这本书虽然比较深，但是其实也没那么难，大家都感觉难啃的原因之一就是中文翻译实在拗口，很多话看中文翻译看半天不知道在说什么，还不如英文原文好理解。java编程思想第四版的中文翻译人员至少给这本书贡献了20%的难度。。。

Reply:
😂